src/mono/mono/metadata/class-init.c

```diff
@@ -605,31 +605,31 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 		switch (m_class_get_byval_arg (klass)->type) {
 			case MONO_TYPE_I1:
 				if (mono_defaults.byte_class)
-					klass->cast_class = mono_defaults.byte_class;
+					mono_defaults.byte_class->cast_class = klass;
```

Comment:
I'm not sure if `byte_class->cast_class` used to be `sbyte_class` before. At least what I can tell from this change. Is this intentional ?

Reply:
all codes still run well if you let sbyte_class->cast_class = byte_class. so it doesn't implement according ECMA 335 but actually work correctly.